### ISSUE TITLE
Smarter DoNotUseInsecureDeserializerWithoutBinderBase

### ISSUE THREAD

**Issue by dotpaul** [dotnet/roslyn-analyzers#1851]
One of the skipped unit tests for `DoNotUseInsecureDeserializerBinaryFormatterWithoutBinder' has:

```csharp
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;

namespace Blah
{
    public class Program
    {
        BinaryFormatter formatter = new BinaryFormatter() { Binder = new MyBinder() };

        public object BfUnsafeDeserialize(byte[] bytes)
        {
            return this.formatter.Deserialize(new MemoryStream(bytes));
        }
    }
}
```

### Ideal behavior
this.formatter.Binder would be detected as set, and no diagnostic is generated.

### Actual behavior
this.formatter.Binder's state is unknown, and generates a diagnostic.

**Comment by mavasani** [comment]
I believe such an analysis can only be done for private fields, by analyzing all the field references and validating it's `Binder` property is never set to another value in the type. Let me know if you had another approach in mind.

**Comment by dotpaul** [comment]
I had no approach in mind, and was thinking that false positives from this case would be rare enough to live with.

**Comment by dotpaul** [comment]
Won't fixing for now.  Will revisit if we get lots of strong feedback.
